File: abc005_c.kt

fun main( args : Array<String> ) {
  val t = readLine()!!.toInt()
  val n = readLine()!!.toInt()
  val ts = readLine()!!.split(" ").map { it.toInt() }
  val m = readLine()!!.toInt()
  val cs = readLine()!!.split(" ").map { it.toInt() }

  val mts = ts.toMutableList()
  cs.map { c ->
    val toTrim = mts.mapIndexed{i,x ->
      Pair(i,x) 
    }.filter { 
      c - it.second  >= 0 && c - it.second <= t 
    }.let { 
      when {
        it.size > 0 -> it.first().first
        else        -> null
      }
    }
    if( toTrim != null ) 
      mts.removeAt(toTrim)
    toTrim
  }.all { 
    it != null
  }.let {
    when(it) {
      true -> "yes"
      else -> "no"
    }.let { println(it) }
  }
}

File: abc005_c.py

T=int(input())
N=int(input())
A=list(map(int,input().split()))
M=int(input())
B=list(map(int,input().split()))


res = []
for j in range(len(B)):
    b = B[j]
    is_match = []
    for i in range(len(A)):
        a = A[i]
        if 0 <= b-a <= T:
            A.pop(i)
            is_match.append(True)
            break
        is_match.append(False)
    res.append(any(is_match) if is_match != [] else False)
#print(res) 
if all(res) is False:
    print("no")
else:
    print("yes")
